q2_demux/test/test_demux.py

```diff
@@ -339,48 +389,249 @@ def test_rev_comp_barcodes(self):
+
+        # sequences in sample2 are correct
+        self._validate_sample_fastq(
+            output_fastq[1][1].open(), self.sequences, [2, 4])
+
+        # sequences in sample3 are correct
+        self._validate_sample_fastq(
+            output_fastq[2][1].open(), self.sequences, [1, 3])
+
+        # sequences in sample4 are correct
+        self._validate_sample_fastq(
+            output_fastq[3][1].open(), self.sequences, [7, 10])
+
+        # sequences in sample5 are correct
+        self._validate_sample_fastq(
+            output_fastq[4][1].open(), self.sequences, [6, 8, 9])
+
+        # manifest is correct
+        act_manifest = list(actual.manifest.view(FastqManifestFormat).open())
+        exp_manifest = ['sample-id,filename,direction\n',
+                        'sample1,sample1_1_L001_R1_001.fastq.gz,forward\n',
+                        'sample3,sample3_2_L001_R1_001.fastq.gz,forward\n',
+                        'sample2,sample2_3_L001_R1_001.fastq.gz,forward\n',
+                        'sample5,sample5_4_L001_R1_001.fastq.gz,forward\n',
+                        'sample4,sample4_5_L001_R1_001.fastq.gz,forward\n']
+        self._compare_manifests(act_manifest, exp_manifest)
 
+        # metadata is correct
+        act_metadata = list(actual.metadata.view(YamlFormat).open())
+        exp_metadata = ["{phred-offset: 33}\n"]
+        self.assertEqual(act_metadata, exp_metadata)
+
+
+class EmpPairedTests(unittest.TestCase, EmpTestingUtils):
+    def setUp(self):
+        self.barcodes = [('@s1/2 abc/2', 'AAAA', '+', 'YYYY'),
+                         ('@s2/2 abc/2', 'TTAA', '+', 'PPPP'),
+                         ('@s3/2 abc/2', 'AACC', '+', 'PPPP'),
+                         ('@s4/2 abc/2', 'TTAA', '+', 'PPPP'),
+                         ('@s5/2 abc/2', 'AACC', '+', 'PPPP'),
+                         ('@s6/2 abc/2', 'AAAA', '+', 'PPPP'),
+                         ('@s7/2 abc/2', 'CGGC', '+', 'PPPP'),
+                         ('@s8/2 abc/2', 'GGAA', '+', 'PPPP'),
+                         ('@s9/2 abc/2', 'CGGC', '+', 'PPPP'),
+                         ('@s10/2 abc/2', 'CGGC', '+', 'PPPP'),
+                         ('@s11/2 abc/2', 'GGAA', '+', 'PPPP')]
+
+        self.forward = [('@s1/1 abc/1', 'GGG', '+', 'YYY'),
+                        ('@s2/1 abc/1', 'CCC', '+', 'PPP'),
+                        ('@s3/1 abc/1', 'AAA', '+', 'PPP'),
+                        ('@s4/1 abc/1', 'TTT', '+', 'PPP'),
+                        ('@s5/1 abc/1', 'ATA', '+', 'PPP'),
+                        ('@s6/1 abc/1', 'TAT', '+', 'PPP'),
+                        ('@s7/1 abc/1', 'CGC', '+', 'PPP'),
+                        ('@s8/1 abc/1', 'GCG', '+', 'PPP'),
+                        ('@s9/1 abc/1', 'ACG', '+', 'PPP'),
+                        ('@s10/1 abc/1', 'GCA', '+', 'PPP'),
+                        ('@s11/1 abc/1', 'TGA', '+', 'PPP')]
+
+        self.reverse = [('@s1/1 abc/1', 'CCC', '+', 'YYY'),
```

Comment:
I couldn't find anything about it either, but the original code indicated that they were the read numbers and stripped them from both the id and description.